openwisp_monitoring/device/tests/test_admin.py

```diff
@@ -480,6 +513,118 @@ def _assert_alertsettings_inline_in_response(response):
+        test_inline_params = {
+            'name': device.name,
+            'organization': str(device.organization.id),
+            'mac_address': device.mac_address,
+            'key': device.key,
+            # metric & alertsettings
+            f'{metric_model_name}-content_type-object_id-TOTAL_FORMS': '1',
+            f'{metric_model_name}-content_type-object_id-INITIAL_FORMS': '0',
+            f'{metric_model_name}-content_type-object_id-MIN_NUM_FORMS': '0',
+            f'{metric_model_name}-content_type-object_id-MAX_NUM_FORMS': '1000',
+            f'{metric_model_name}-content_type-object_id-0-configuration': 'iperf',
+            f'{metric_model_name}-content_type-object_id-0-id': '',
+            f'{metric_model_name}-content_type-object_id-0-alertsettings-TOTAL_FORMS': '1',
+            f'{metric_model_name}-content_type-object_id-0-alertsettings-INITIAL_FORMS': '0',
+            f'{metric_model_name}-content_type-object_id-0-alertsettings-MIN_NUM_FORMS': '0',
+            f'{metric_model_name}-content_type-object_id-0-alertsettings-MAX_NUM_FORMS': '1',
+            f'{metric_model_name}-content_type-object_id-0-alertsettings-0-is_active': 'on',
+            f'{metric_model_name}-content_type-object_id-0-alertsettings-0-custom_operator': '<',
+            f'{metric_model_name}-content_type-object_id-0-alertsettings-0-custom_threshold': '9',
+            f'{metric_model_name}-content_type-object_id-0-alertsettings-0-custom_tolerance': '1800',
+            f'{metric_model_name}-content_type-object_id-0-alertsettings-0-id': '',
+            f'{metric_model_name}-content_type-object_id-0-alertsettings-0-metric': '',
+        }
+        # Only general metrics (clients & traffic) are present
+        self.assertEqual(Metric.objects.count(), 2)
+        self.assertEqual(AlertSettings.objects.count(), 0)
+
+        def _reset_alertsettings_inline():
+            Metric.objects.all().delete()
+            AlertSettings.objects.all().delete()
+
+        # Delete AlertSettingsInline objects before any subTests
+        _reset_alertsettings_inline()
+
+        def _assert_alertsettings_inline(response, operator, threshold, tolerance):
+            self.assertEqual(response.status_code, 302)
+            self.assertEqual(Metric.objects.count(), 1)
+            self.assertEqual(AlertSettings.objects.count(), 1)
+            alertsettings = AlertSettings.objects.first()
+            self.assertEqual(alertsettings.operator, operator)
+            self.assertEqual(alertsettings.threshold, threshold)
+            self.assertEqual(alertsettings.tolerance, tolerance)
+
+        with self.subTest('Test alert settings inline when all fields are provided'):
+            self._device_params.update(test_inline_params)
+            response = self.client.post(url, self._device_params)
+            _assert_alertsettings_inline(response, '<', 9, 1800)
+            _reset_alertsettings_inline()
```

Comment:
@Aryamanz29 the resetting should be done outside of the sub-test.

Reply:
ahh 😅